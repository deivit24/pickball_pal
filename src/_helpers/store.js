import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
