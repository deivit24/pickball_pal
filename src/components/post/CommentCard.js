// React
import React from 'react';
// Redux
import { useSelector, useDispatch } from 'react-redux';
// React Router Dom
import { Link } from 'react-router-dom';
import { Avatar, Grid, Paper, Button, Icon } from '@material-ui/core';
// Moment
import Moment from 'react-moment';
// Delete Comment Action
import { deleteComment } from '../../_actions/post';

const CommentCard = ({
  postId,
  comment: { _id, text, first_name, last_name, avatar, user, date },
}) => {
  const dispatch = useDispatch();

  const auth = useSelector((state) => state.auth);

  return (
    <Paper style={{ padding: '20px', marginBottom: '10px' }}>
      <Grid container wrap="nowrap" spacing={2}>
        <Link to={`/profile/${user}`}>
          <Grid item>
            <Avatar alt="avatar" src={avatar} />
          </Grid>
        </Link>
        <Grid justifycontent="left" item xs zeroMinWidth>
          <h4 style={{ margin: 0, textAlign: 'left' }}>
            {first_name} {last_name}
          </h4>
          <p style={{ textAlign: 'left' }}>{text}</p>
          <div className="row">
            <p style={{ textAlign: 'left', color: 'gray' }}>
              Posted <Moment to={date} />
            </p>
            {!auth.loading && user === auth.user._id && (
              <Button
                onClick={(e) => dispatch(deleteComment(postId, _id))}
                color="secondary"
                style={{ marginLeft: 'auto' }}
              >
                <Icon className="fas fa-trash"></Icon>
              </Button>
            )}
          </div>
        </Grid>
      </Grid>
    </Paper>
  );
};

export default CommentCard;
