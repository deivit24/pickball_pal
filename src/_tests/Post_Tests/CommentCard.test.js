import React from 'react';
import { render } from '@testing-library/react';
import CommentCard from '../../components/post/CommentCard';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing CommentCard

describe('Testing the CommentCard', () => {
  let postId = '0001';
  let comment = {
    _id: '00002',
    text: 'hey this is a comment',
    first_name: 'david',
    last_name: 'salazar',
    avatar:
      'https://icon-library.com/images/default-user-icon/default-user-icon-4.jpg',
    user: '00003',
    date: Date.now(),
  };
  it('it renders CommentCard without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <CommentCard postId={postId} comment={comment} />
        </Provider>
      </MemoryRouter>
    );
  });

  // Wont work because there is a Date.now() and it changes from snapshot
  // it('matches snapshot', () => {
  //   const { asFragment } = render(
  //     <MemoryRouter>
  //       <Provider store={store}>
  //         <CommentCard postId={postId} comment={comment} />
  //       </Provider>
  //     </MemoryRouter>
  //   );
  //   expect(asFragment()).toMatchSnapshot();
  // });
});
