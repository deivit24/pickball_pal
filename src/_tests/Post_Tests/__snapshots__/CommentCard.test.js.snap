// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the CommentCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    style="padding: 20px; margin-bottom: 10px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap"
    >
      <a
        href="/profile/00003"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle"
          >
            <img
              alt="avatar"
              class="MuiAvatar-img"
              src="https://icon-library.com/images/default-user-icon/default-user-icon-4.jpg"
            />
          </div>
        </div>
      </a>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-true"
        justifycontent="left"
      >
        <h4
          style="margin: 0px; text-align: left;"
        >
          david salazar
        </h4>
        <p
          style="text-align: left;"
        >
          hey this is a comment
        </p>
        <div
          class="row"
        >
          <p
            style="text-align: left; color: gray;"
          >
            Posted 
            <time
              datetime="1601072230177"
            >
              a few seconds ago
            </time>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
